packages/server/src/profile/profile.service.ts

```diff
@@ -0,0 +1,49 @@
+import { KhouryProfCourse } from '@koh/common';
+import { Injectable } from '@nestjs/common';
+import { LastRegistrationModel } from 'login/last-registration-model.entity';
+import { LoginCourseService } from 'login/login-course.service';
+import { ProfSectionGroupsModel } from 'login/prof-section-groups.entity';
+import { Connection } from 'typeorm';
+
+@Injectable()
+export class ProfileService {
+  constructor(
+    private connection: Connection,
+    private loginCourseService: LoginCourseService,
+  ) {}
+
+  public async getPendingCourses(userId: number): Promise<KhouryProfCourse[]> {
+    const profCourses = await ProfSectionGroupsModel.findOne({
+      where: {
+        profId: userId,
+      },
+    });
+    const lastRegistered = await LastRegistrationModel.findOne({
+      where: {
+        profId: userId,
+      },
+    });
+
+    if (!profCourses) return; // not a professor
```

Comment:
should the pending courses not exist in the profile payload at all if the professor has already registered? Or should we use the empty list? 

I think this would affect how we want to redirect

Reply:
Ermm yeah we were just gonna check for both the undefined and empty array case on the FE haha.. partially bc its possible that the prof hasn't already registered for this sem, but all their courses have actually been registered already by other profs (in which case the logic ends up generating an empty array as well)

We can add some more logic here to ensure the function only ever returns either undefined or empty array if you feel strongly about it, though.

Reply:
Ahh ok yea that would work -- personally I think it's better to have the pending courses be populated or nonexistent (one less check on the frontend lol). However it's fine either way